src/vma/dev/net_device_table_mgr.cpp

```diff
@@ -323,10 +333,74 @@ void net_device_table_mgr::verify_ipoib_mode(struct ifaddrs* ifa)
 		vlog_printf(VLOG_WARNING,"This option in no longer needed in this version\n");
 		vlog_printf(VLOG_WARNING,"Please refer to Release Notes for more information\n");
 		vlog_printf(VLOG_WARNING,"************************************************************************\n");
+		return 0;
 	}
 	else {
 		ndtm_logdbg("verified interface '%s' is running with umcast disabled", ifa->ifa_name);
 	}
+	return 1;
+}
+
+bool net_device_table_mgr::check_ib_device_supported(const char* ifname) {
+	if (check_device_exist(ifname, CX4_DRIVER_PATH)) {
+		vlog_printf(VLOG_WARNING,"************************************************************************\n");
+		vlog_printf(VLOG_WARNING,"VMA doesn't support IB on CX4\n");
+		vlog_printf(VLOG_WARNING,"All traffic over interface %s will not be offloaded!\n", ifname);
+		vlog_printf(VLOG_WARNING,"************************************************************************\n");
+		return 0;
+	} else {
+		return check_qp_creation(ifname, IBV_QPT_UD);
+	}
+}
+
+bool net_device_table_mgr::check_qp_creation(const char* ifname, enum ibv_qp_type qp_type) {
+	int num_devices = 0;
+
+	struct ibv_qp_init_attr qp_init_attr;
+	memset(&qp_init_attr, 0, sizeof(qp_init_attr));
+
+	vma_ibv_cq_init_attr attr;
+	memset(&attr, 0, sizeof(attr));
+
+	qp_init_attr.cap.max_send_wr = safe_mce_sys().tx_num_wr;
+	qp_init_attr.cap.max_recv_wr = safe_mce_sys().rx_num_wr;
+	qp_init_attr.cap.max_inline_data = safe_mce_sys().tx_max_inline;
+	qp_init_attr.cap.max_send_sge = MCE_DEFAULT_TX_NUM_SGE;
+	qp_init_attr.cap.max_recv_sge = MCE_DEFAULT_RX_NUM_SGE;
+	qp_init_attr.sq_sig_all = 0;
+	qp_init_attr.qp_type = qp_type;
+
+	//find ib_cxt
+	char base_ifname[IFNAMSIZ];
+	get_base_interface_name((const char*)(ifname), base_ifname, sizeof(base_ifname));
+	struct ibv_context** pp_ibv_context_list = rdma_get_devices(&num_devices);
+	char resource_path[256];
+	sprintf(resource_path, VERBS_DEVICE_RESOURCE_PARAM_FILE, base_ifname);
+	char sys_res[1024] = {0};
+	priv_read_file(resource_path, sys_res, 1024);
```

Comment:
please consider using priv_safe_read_file(resource_path, sys_res, sizeof(sys_res));
in case returned buffer is null terminated string.